spec/operators/scan-spec.ts

```diff
@@ -249,7 +249,7 @@ describe('Observable.prototype.scan', () => {
   it('should accept R typed reducers', () => {
```

Comment:
isn't it for `T` only overload testing? 
```
scan<T>(this: Observable<T>, accumulator: (acc: T, value: T, index: number) => T, seed?: T)
```

Reply:
Well, the test is called `should accept R typed reducers`. I'm assumed that meant we're explicitly testing the case where we have two different types for `acc` and the source. I'm guessing that based on the other tests in the file, i.e, there's another one for `T typed reducers` and `Array typed reducers'

Reply:
I think we can't just trust test name and test cases here at all, since from first it was testing `reduce` anyway 😅  I was just guessing by usage of code here.

Reply:
Because of the way the overloads and the tests are set-up, in the actual test, both `acc` and `value` are T. Which means acc isn't really inheriting what `R` is supposed to be.

Reply:
Maybe we should leave this as-is for now, and i'll propose some changes to these tests in https://github.com/ReactiveX/rxjs/pull/2897 after we get this in ?